application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/views/stacktrace/StacktraceView.java

```diff
@@ -960,6 +974,32 @@ public String getText(Object element) {
 		}
 	};
 
+	private final ColumnLabelProvider durationLabelProvider = new ColumnLabelProvider() {
+		@Override
+		public String getText(Object element) {
+			StacktraceFrame frame = (StacktraceFrame) element;
+			IQuantity duration = getDurationCount(frame.getItems());
+			return formatDuration(duration.longValue(), duration.getUnit().getIdentifier());
+		}
+	};
+
+	private String formatDuration(long duration, String unit) {
+		Duration rawDuration = null;
+		String formattedTime = "NA";
+		if (unit.equalsIgnoreCase("ns"))
```

Comment:
Also are these the only possible time formats? I was playing around with some of recordings and on the threads pages the string unit sometimes comes in as ticks, so the duration just lists "NA" in the column (but the percentage by duration column still works).

Reply:
Implemented the first review comment.
I have implemented as per the ones available in java.time.Duration class and the general units which I could see with the recordings I have. I couldn't find ticks in Duration class. May be we can modify the code in future as and when required with a simple P4 JIRA Ticket.

Reply:
Is that the correct duration? I was thinking that it might not be related to the duration not of java.time.Duration, but to the duration relating to jdk.jfr (timespan?).

The `formatDuration()` here gets the duration from `StacktraceView.getDurationCount()` [0], which looks up the duration in the form of a timespan [1].

Timespans [2] have fields for microseconds, milliseconds, nanoseconds, seconds, and ticks, which explains what I'm seeing one of my recordings.

[0] https://github.com/openjdk/jmc/blob/master/application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/views/stacktrace/StacktraceView.java#LL943C1-L943C2
[1] https://github.com/openjdk/jmc/blob/master/core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/JfrAttributes.java#L108
[2] m/en/java/javase/17/docs/api/jdk.jfr/jdk/jfr/Timespan.html